src/QirRuntime/samples/StandaloneInputReference/qir-driver.cpp

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "CLI11.hpp"
+#include "CoreTypes.hpp"
+#include "QirContext.hpp"
+#include "QirTypes.hpp"
+#include "QuantumApi_I.hpp"
+#include "SimFactory.hpp"
+#include "quantum__qis_internal.hpp"
+#include "quantum__rt.hpp"
+#include <cctype>
+#include <iostream>
+#include <map>
+#include <vector>
+
+using namespace Microsoft::Quantum;
+using namespace std;
+
+// This is the function corresponding to the QIR entry-point.
+extern "C" int64_t Quantum__StandaloneSupportedInputs__ExerciseInputs__body( // NOLINT
+    int64_t intValue,
+    double doubleValue);
+
+map<string, PauliId> PauliMap{
+    {"PauliI", PauliId::PauliId_I},
+    {"PauliX", PauliId::PauliId_X},
+    {"PauliY", PauliId::PauliId_Y},
+    {"PauliZ", PauliId::PauliId_Z}};
+
+map<string, bool> ResultMap{
+    {"0", false},
+    {"Zero", false},
+    {"1", true},
+    {"One", true}};
+
+int main(int argc, char* argv[])
+{
+    CLI::App app("QIR Standalone Entry Point Inputs Reference");
+
+    // Initialize simulator.
+    unique_ptr<ISimulator> sim = CreateFullstateSimulator();
+    QirContextScope qirctx(sim.get(), false /*trackAllocatedObjects*/);
+
+    // Add the --simulation-output and --operation-output options.
+    // N.B. These options should be present in all standalone drivers.
+    string simulationOutputFile;
+    CLI::Option* simulationOutputFileOpt = app.add_option(
+        "-s,--simulation-output", simulationOutputFile,
+        "File where the output produced during the simulation is written");
+
+    string operationOutputFile;
+    CLI::Option* operationOutputFileOpt = app.add_option(
+        "-o,--operation-output", operationOutputFile, "File where the output of the Q# operation is written");
+
+    // Add the options that correspond to the parameters that the QIR entry-point needs.
+    // Option for a Q# Int type.
+    int64_t intValue = 0;
+    app.add_option("--int-value", intValue, "An integer value")->required();
+
+    // Option for a Q# Double type.
+    double_t doubleValue = 0.0;
+    app.add_option("--double-value", doubleValue, "A double value")->required();
```

Comment:
The fragment `->required()` makes me think that the test will not run if the `--double-value` is not specified in the command line. Is that the case? Does the test really require that value to run? Some default value like `0.0` will not work?
If that `--double-value` is not specified then what will the test return to command line? Will that be interpreted as the _failure_ of the test?

Reply:
Sorry. Looks like this is not a test, this is a sample.